### DISCUSSION TITLE
How are you using the toolkit?

### DISCUSSION THREAD

**Discussion by smr99** [smr99/lego-hub-tk#25] [General]
When I created this toolkit a year ago, one of my goals was to be able to seamlessly switch between wired USB and Bluetooth connectivity.  This led to a fairly complex design of "ConnectionManager" with auto-discovery for each type of connection, and failover.

It worked reasonably well for me on linux.   But from the patches I've received from folks on windows or mac, it seems that either this functionality doesn't work or isn't desired.

So I'm curious how you are using the toolkit.  
* Do you only use run_command.py to load code?
* Do you use either of the GUI tools hubstatus.py or hubcontrol.py?
* Are you writing your own scripts using the toolkit?
* Do you use Bluetooth?  
* Is failover from USB to BT important to you?

Are there issues or stumbling blocks that make the toolkit harder to use than it needs to be?

**Comment by SpudGunMan** [comment]
being able to choose USB or BT is important auto detect isn't important IMO.
I wanted to have a copy of the project working to use in robot ideas and using a 7" screen with a Pi for status and UI control and variable gathering for data from the pc realtime feedback for robot control etc blah blah

when you asked about the toolkit I haven't dug that deep yet but around the documentation on API and use of the project as a library I haven't fully uncovered yet, for example I do like the demo GUI applications they are nice. I just havent spent the time personally yet to find if I can use a not QT5 connectivity with your library and the robot

**Reply by smr99** [reply]
Just pushed an update that makes the connection fully specifiable using runtime configuration.  Let me know what you think!

**Reply by SpudGunMan** [reply]
looks good I will dig in more with this soon!
